source/test/lysis_mode_test/LysisDataNodes.test.cc

```diff
@@ -59,4 +59,215 @@ TEST_CASE("GetCFUDataNode", "[lysis]"){
+        p->SetLysisChance(lysis_chances[i+3]);
+        h->AddSymbiont(p);
+      }
+      w.Update();
+      //check they were tracked correctly
+      THEN("their average lysis chances are tracked"){
+        REQUIRE(lysis_chance_data_node.GetMean() < (expected_av + 0.0001));
+        REQUIRE(lysis_chance_data_node.GetMean() > (expected_av - 0.0001));
+      }
+    }
+  }
+}
+
+TEST_CASE("GetInductionChanceDataNode", "[lysis]"){
+  GIVEN( "a world" ) {
+    emp::Random random(17);
+    SymConfigBase config;
+    int int_val = 0;
+    LysisWorld w(random);
+    w.Resize(4);
+    w.SetFreeLivingSyms(1);
+    config.SYM_LIMIT(3);
+
+    size_t max_bin = 10;
+
+    emp::DataMonitor<double,emp::data::Histogram>& induction_chance_data_node = w.GetInductionChanceDataNode();
+    induction_chance_data_node.SetupBins(-1.0, 1.0, max_bin);
+    REQUIRE(std::isnan(induction_chance_data_node.GetMean()));
+    for(size_t i = 0; i < max_bin; i++){
+      REQUIRE(induction_chance_data_node.GetHistCounts()[i] == 0);
+    }
+
+    WHEN("syms are added"){
+      //setup
+      size_t num_syms = 6;
+      double induction_chances[6] = {0, 0.13, 0.26, 0.45, 0.89, 1.0};
+      double expected_av = 0.455;
+
+      //insert organisms
+      for(size_t i = 0; i < (num_syms / 2); i++){
+        Phage *p = new Phage(&random, &w, &config, int_val);
+        p->SetInductionChance(induction_chances[i]);
+        w.AddOrgAt(p, emp::WorldPosition(0,i));
+      }
+      Bacterium *h = new Bacterium(&random, &w, &config, int_val);
+      w.AddOrgAt(h, 0);
+      for(size_t i = 0; i < (num_syms / 2); i++){
+        Phage *p = new Phage(&random, &w, &config, int_val);
+        p->SetInductionChance(induction_chances[i+3]);
+        h->AddSymbiont(p);
+      }
+      w.Update();
+      //check they were tracked correctly
+      THEN("their average induction chances are tracked"){
+        REQUIRE(induction_chance_data_node.GetMean() < (expected_av + 0.0001));
+        REQUIRE(induction_chance_data_node.GetMean() > (expected_av - 0.0001));
+      }
+    }
+  }
+}
+
+TEST_CASE("GetBurstSizeDataNode", "[lysis]"){
+  GIVEN( "a world" ) {
+    emp::Random random(17);
+    SymConfigBase config;
+    int int_val = 0;
+    size_t burst_time = 3;
+    LysisWorld w(random);
+    w.Resize(4);
+    w.SetFreeLivingSyms(1);
+    config.LYSIS(1);
+    config.LYSIS_CHANCE(1);
+    config.BURST_TIME(burst_time);
+
+    emp::DataMonitor<double>& burst_size_data_node = w.GetBurstSizeDataNode();
+    REQUIRE(std::isnan(burst_size_data_node.GetMean()));
+
+    WHEN("bacteria lyse"){
+      size_t burst_sizes[4] = {4,5,6,7};
+      double expected_av = 5.5;
+      for(int i = 0; i < 4; i++){ // populate world with 4 bacteria
+        //each of which has a different burst size
+        Bacterium *h = new Bacterium(&random, &w, &config, int_val);
+        Phage *p = new Phage(&random, &w, &config, int_val);
+        for(size_t j = 0; j < burst_sizes[i]; j++) h->AddReproSym(p->reproduce());
```

Comment:
I'm not a fan of single line for loops, let's expand this and break the p->reproduce into multiple lines so that it's clear what is going on here